writeaheadlog.go

```diff
@@ -242,6 +243,16 @@ func (w *WAL) recoverWAL(data []byte) ([]Update, error) {
 			firstPage.nextPage = &nextDiskPage.page
 		}
 
+		// Check if the pages of the transaction form a loop
```

Comment:
@DavidVorick should revisit this during the final run-through and explain in the comment what circumstances could lead to a cycle